src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/KeyInfoX509Data.cs

```diff
@@ -160,6 +160,15 @@ public void AddIssuerSerial(string issuerName, string serialNumber)
             if (!BigInteger.TryParse(serialNumber, NumberStyles.AllowHexSpecifier, NumberFormatInfo.CurrentInfo, out h))
                 throw new ArgumentException(SR.Cryptography_Xml_InvalidX509IssuerSerialNumber, nameof(serialNumber));
 
+            // NetFx compat: .NET Framework treats the input as unsigned and we need to write down the X509SerialNumber
+            // as a positive number.
+            if (h < BigInteger.Zero)
+            {
+                byte[] bytes = h.ToByteArray();
```

Comment:
Since this is an uncommon path, and need to work in .NET Standard, I didn't think it was worthwhile to special-case NETCOREAPP which gives us `isUnsigned`.